packages/@react-stately/utils/src/useControlledState.ts

```diff
@@ -47,15 +47,10 @@ export function useControlledState<T>(
       // this will call our useState setState with a function as well which invokes myFunc and calls onChange with the value from myFunc
       // if we're in an uncontrolled state, then we also return the value of myFunc which to setState looks as though it was just called with myFunc from the beginning
       // otherwise we just return the controlled value, which won't cause a rerender because React knows to bail out when the value is the same
-      let updateFunction = (oldValue, ...functionArgs) => {
-        let interceptedValue = value(isControlled ? stateRef.current : oldValue, ...functionArgs);
-        onChangeCaller(interceptedValue, ...args);
-        if (!isControlled) {
-          return interceptedValue;
-        }
-        return oldValue;
-      };
-      setStateValue(updateFunction);
+      let oldValue = stateRef.current;
+      let interceptedValue = value(oldValue);
```

Comment:
This line doesn't seem like it is the same. I'm having trouble figuring out what the expected different between oldValue and stateRef.current in the original code would be and what they are virtually the same in the current code. 

In addition, the comment for this section of code seems out of date.

Reply:
ah, yep, that comment is now out of date
and yes, it's not quite the same as before, which is why we've pulled it out into its own PR, the issue was that the old way of doing it had a side effect
it used to update the state, once that update was complete, it would call onChange
now it queues the state update and calls onChange, possibly before the state has been committed